### ISSUE TITLE
[Feature]: Clippy lint undocumented unsafe blocks

### ISSUE THREAD

**Issue by haraldh** [enarx/enarx#1661]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

For better review and a later audit process, it is critical to have safety comments for all unsafe blocks.

To make those blocks as small as possible, unsafe functions should allow unsafe blocks in the inside also.

### Suggestions for a technical implementation

Add
```
#![deny(clippy::missing_safety_doc)]
#![deny(clippy::undocumented_unsafe_blocks)]
#![deny(unsafe_op_in_unsafe_fn)]
```

**Comment by bstrie** [comment]
Yes, having every unsafe block documented will be mandatory if we ever hope to pass an actual audit. In general I would also like to see safety comments be more rigorous and standardized, but that can happen in a future effort.

Also turning on more clippy lints in general is something that we should pursue.
